wai-aria/role/invalid-roles.html

```diff
@@ -11,26 +11,109 @@
+<nav role="&nbsp" data-testname="nav with non-breaking space (nbsp) as role" data-expectedrole="navigation" class="ex">x</nav>
+<nav role="&#20" data-testname="nav with standard space (nbsp) as role" data-expectedrole="navigation" class="ex">x</nav>
+
+<!-- Whitespace tests with <span> (including line breaks, tabs, zero-width space, braille space, non-breaking space, standard space) -->
+<span role=" " data-testname="span with empty character as role" class="ex-generic">x</span>
+<span role="&#13" data-testname="span with line break (&#13) character as role" class="ex-generic">x</span>
+<span role="&#9" data-testname="span with tab (&#9) as role (should be treated as whitespace)" class="ex-generic">x</span>
+<span role="&#8203" data-testname="span with zero-width space as role (should be treated as whitespace)" class="ex-generic">x</span>
+<span role="&#10240" data-testname="span with braille space (10240) as role" class="ex-generic">x</span>
+<span role="&#x2800" data-testname="span with braille space (x2800) as role" class="ex-generic">x</span>
+<span role="&nbsp" data-testname="span with non-breaking space (nbsp) as role" class="ex-generic">x</span>
+<span role="&#20" data-testname="span with standard space (nbsp) as role" class="ex-generic">x</span>
+
+<!-- Diacritics with <a> -->
+<a role="&#771" data-testname="link with role set to tilde diacritic" data-expectedrole="link" class="ex">x</a>
+<a role="foo&#771" data-testname="link with role foo that has tilde diacritic" data-expectedrole="link" class="ex">x</a>
+<a role="&#773" data-testname="link with role set to overline diacritic" data-expectedrole="link" class="ex">x</a>
+<a role="foo&#773" data-testname="link with role foo that has overline diacritic" data-expectedrole="link" class="ex">x</a>
+<a role="&#813" data-testname="link with role set to circumflex diacritic below" data-expectedrole="link" class="ex">x</a>
+<a role="foo&#813" data-testname="link with role foo that has circumflex diacritic below" data-expectedrole="link" class="ex">x</a>
+<a role="button&#771" data-testname="link with role set to button with tilde diacritic" data-expectedrole="link" class="ex">x</a>
+
+<!-- Diacritics with <div> -->
+<div role="&#771" data-testname="div with role set to tilde diacritic" class="ex-generic">x</div>
+<div role="foo&#771" data-testname="div with role foo that has tilde diacritic" class="ex-generic">x</div>
+<div role="button&#771" data-testname="div with role set to button with tilde diacritic" class="ex-generic">x</div>
+
+<!-- Zero-width joiners (e.g., ZWJ like emoji variants use) with <button> -->
+<button role="&#8205" data-testname="button with role set to zero-width joiner" data-expectedrole="button" class="ex">x</button>
+<button role="foo&#8205" data-testname="button with role set to foo with zero-width joiner" data-expectedrole="button" class="ex">x</button>
```

Comment:
```suggestion
<!-- [sic] role="‍" includes a ZWJ], and role="link‍" ends with an invisible ZWJ. Use caution when editing. -->
<button role="‍" data-testname="button with role set to zero-width joiner" data-expectedrole="button" class="ex">x</button>
<button role="link‍" data-testname="button with role set to foo with zero-width joiner" data-expectedrole="button" class="ex">x</button>
```

Reply:
I think I did that correctly. Hard to tell in the GH interface though.

Reply:
copying in between these brackets too [‍]... if you have trouble verifying offline, I have some tips.

Reply:
https://unicode-explorer.com/c/200D